src/components/common/Layout/components/Account/components/Notifications/components/xmtp.ts

```diff
@@ -0,0 +1,142 @@
+import { useToast } from "@chakra-ui/react"
+import { Client, useCanMessage } from "@xmtp/react-sdk"
+import useUser from "components/[guild]/hooks/useUser"
+import useShowErrorToast from "hooks/useShowErrorToast"
+import { useCallback, useEffect, useState } from "react"
+import { useFetcherWithSign } from "utils/fetcher"
+import { useAccount, useWalletClient } from "wagmi"
+
+export const useSaveXmtpKeys = () => {
+  const fetcherWithSign = useFetcherWithSign()
+  const { data: signer } = useWalletClient()
+
+  const { id } = useUser()
+
+  return useCallback(
+    async () =>
+      Client.getKeys(signer, { env: "production" }).then(async (key) =>
+        fetcherWithSign([
+          id ? `/v2/users/${id}/keys` : undefined,
+          {
+            body: {
+              key: Buffer.from(key).toString("binary"),
+              service: "XMTP",
+            },
+          },
+        ])
+      ),
+    [id, signer]
+  )
+}
+
+export const useDeleteXmtpKeys = () => {
+  const fetcherWithSign = useFetcherWithSign()
+  const { id } = useUser()
+
+  return async () =>
+    fetcherWithSign([
+      id ? `/v2/users/${id}/keys/1` : undefined,
+      { method: "DELETE" },
+    ])
+}
+
+export const useGetXmtpKeys = () => {
```

Comment:
The hook returns the data/error & the loading state, so let's call it `useUsersXMTPKeys`, that's a more logical name in my opinion